### PR TITLE
Autodiscover snyk organizations

### PR DESCRIPTION
**Pull request by PeyGis**
# Description
We got a feedback from a customer that it’s hard to create an exporter for each organization they have in snyk. And they would prefer to bring all the orgaanizations  on a specific group.

What -
Why -
How - Using config mapping and the token provided in the ocean integration, we fetch all the organizations that are associated with the given token. We then fetch all snyk data (projects, targets, issues) for all organizations. We also make it possible for the user to filter out any specific group, instead of all organizations.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: integrations/snyk/snyk/client.py [hunk 1]
```diff
@@ -290,19 +301,67 @@ async def enrich_project(self, project: dict[str, Any]) -> dict[str, Any]:
...
+                response = await self._send_api_request(url=url)
+                organization_ids_in_group = [org["id"] for org in response["orgs"]]
+                logger.info(
+                    f"Fetched {len(organization_ids_in_group)} organizations in group: {group_id}. Organization IDs: {str(organization_ids_in_group)}"
+                )
+                all_organizations.extend(organization_ids_in_group)
+
+            logger.info(
+                f"Fetched {len(all_organizations)} organizations for the given groups. All organization IDs: {str(all_organizations)}"
+            )
```
Comment (Tankilevitch) [inline integrations/snyk/snyk/client.py:351]: this can be done in parallel, instead of one by one

File: integrations/snyk/snyk/client.py [hunk 2]
```diff
@@ -290,19 +301,67 @@ async def enrich_project(self, project: dict[str, Any]) -> dict[str, Any]:
...
-            url=snyk_webhook_url, method="POST", json_data=body
-        )
+            body = {"url": app_host_webhook_url, "secret": self.webhook_secret}
+            logger.info(f"Creating webhook subscription for organization: {org_id}")
+            await self._send_api_request(
+                url=snyk_webhook_url, method="POST", json_data=body
+            )
+
+    async def _get_organizations_in_group(self) -> list[Any]:
+        # Check if the result is already cached
```
Comment (Tankilevitch) [inline integrations/snyk/snyk/client.py:326]: this function supports bringing orgs from multiple groups, while the method name relates to one group, lets adjust the naming

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Some changes are still required, overall nice job 🌊

File: integrations/snyk/main.py [hunk 3]
```diff
@@ -62,8 +68,7 @@ async def on_projects_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
         logger.debug(
             f"Received batch with {len(projects)} projects, getting their issues"
         )
-        semaphore = asyncio.Semaphore(5)
-        tasks = [process_project_issues(semaphore, project) for project in projects]
+        tasks = [process_project_issues(project) for project in projects]
```
[the remark this answers is not in the document]
Comment (PeyGis) [reply integrations/snyk/main.py:71]: I can bump it to 20 then

File: integrations/snyk/.port/spec.yaml [hunk 4]
```diff
@@ -14,17 +15,26 @@ configurations:
...
     required: false
     default: https://api.snyk.io
+    description: The base URL of the Snyk API. Defaults to https://api.snyk.io
   - name: organizationId
     type: string
-    required: true
+    required: false
+    description: The unique identifier for the specific organization. Fetches data for a single organization when provided
+  - name: groups
+    type: string
```
[the remark this answers is not in the document]
Comment (PeyGis) [reply integrations/snyk/.port/spec.yaml:34]: The integration docs shows only these types that's why I went for the string

Available values: string, number, boolean, object, url

[the remark this answers is not in the document]
Comment (Tankilevitch) [reply integrations/snyk/.port/spec.yaml:34]: I think we missed it in the docs, but I know that the azure integration uses the list type, can you check it please? if that works I prefer us using list which is more straight forward to the user.

File: integrations/snyk/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tankilevitch) [reply integrations/snyk/main.py]: I found this - https://snyk.docs.apiary.io/#introduction/errors 
so 20 I guess is alright :)

File: integrations/snyk/.port/spec.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (PeyGis) [reply integrations/snyk/.port/spec.yaml:34]: Sure

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good

I still think we need to handle the organizations response in a different way, instead of one time bringing the organizations from the rest api and sometimes from the groups api and those be strict with the data that is returned.
Let's pull all the organizations from the API, and filter the unrelated ones from the organizations list that is being returned from the group API.

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Pull request merged**